dot/parachain/statement-distribution/state_v2.go

```diff
@@ -0,0 +1,206 @@
+}
+
+// isDisabled returns `true` if the given validator is disabled in the context of the relay parent.
+func (p *perRelayParentState) isDisabled(vIdx parachaintypes.ValidatorIndex) bool {
+	_, ok := p.disabledValidators[vIdx]
+	return ok
+}
+
+func (p *perRelayParentState) disableBitmask(group []parachaintypes.ValidatorIndex) (parachaintypes.BitVec, error) {
+	disableBm := make([]bool, len(group))
+	for idx, v := range group {
+		disableBm[idx] = p.isDisabled(v)
+	}
+
+	bm, err := parachaintypes.NewBitVec(disableBm)
+	if err != nil {
+		logger.Criticalf("cannot create a bitvec: %s", err.Error())
+	}
+
+	return bm, err
+}
+
+type localValidatorStore struct {
+	gridTracker any // TODO: use GridTracker implementation
+	active      *activeValidatorState
+}
+
+type activeValidatorState struct {
+	index          parachaintypes.ValidatorIndex
+	groupIndex     parachaintypes.GroupIndex
+	assignments    []parachaintypes.ParaID
+	clusterTracker any // TODO: use cluster tracker implemenetation
+}
+
+type perSessionState struct {
+	sessionInfo parachaintypes.SessionInfo
+	groups      *groups
+	authLookup  map[parachaintypes.AuthorityDiscoveryID]parachaintypes.ValidatorIndex
+	gridView    any // TODO: use SessionTopologyView from statement-distribution grid
+
+	// when localValidator is nil means it is inactive
+	localValidator     *parachaintypes.ValidatorIndex
+	allowV2Descriptors bool
+}
+
+func newPerSessionState(sessionInfo parachaintypes.SessionInfo,
+	keystore keystore.Keystore,
+	backingThreshold uint32,
+	allowV2Descriptor bool,
+) *perSessionState {
+	authlookup := make(map[parachaintypes.AuthorityDiscoveryID]parachaintypes.ValidatorIndex)
+	for idx, ad := range sessionInfo.DiscoveryKeys {
+		authlookup[ad] = parachaintypes.ValidatorIndex(idx)
+	}
+
+	var localValidator *parachaintypes.ValidatorIndex
+	validatorPk, validatorIdx := parachainutil.SigningKeyAndIndex(sessionInfo.Validators, keystore)
+	if validatorPk != nil {
+		localValidator = &validatorIdx
+	}
+
+	return &perSessionState{
+		sessionInfo:        sessionInfo,
+		groups:             newGroups(sessionInfo.ValidatorGroups, backingThreshold),
+		authLookup:         authlookup,
+		localValidator:     localValidator,
+		allowV2Descriptors: allowV2Descriptor,
+		gridView:           nil,
+	}
+}
+
+// supplyTopology sets the topology for the session and updates the local validator
+// Note: we use the local index rather than the `perSessionState.local_validator` as the
+// former may be `Some` when the latter is `None`, due to the set of nodes in
+// discovery being a superset of the active validators for consensus.
+func (s *perSessionState) supplyTopology(topology *grid.SessionGridTopology, localIdx *parachaintypes.ValidatorIndex) {
+	// TODO #4373: implement once buildSessionTopology is done
+	// gridView := buildSessionTopology(
+	// 	s.sessionInfo.ValidatorGroups,
+	// 	topology,
+	// 	localIdx,
+	// )
+
+	// s.gridView = gridView
+
```

Comment:
Why do we have this commented code? Can we remove this?

Reply:
I will keep commented since this code will be production code once we implement `buildSessionTopology` in the next few days.